modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/IgniteSqlValidator.java

```diff
@@ -479,6 +487,61 @@ private void checkTypesInteroperability(SqlValidatorScope scope, SqlNode expr) {
                     throw SqlUtil.newContextException(expr.getParserPosition(), ex);
                 }
             }
+
+            literalCanFitType(expr, returnType);
+        }
+    }
+
+    /** Check literal can fit to declared exact numeric type, work only for single literal. */
+    private static void literalCanFitType(SqlNode expr, RelDataType toType) {
+        if (INT_TYPES.contains(toType.getSqlTypeName())) {
+            LiteralExtractor litExtractor = new LiteralExtractor();
```

Comment:
There is no need to create a new instance for every expression, because `SqlValidator` is stateful and `LiteralExtractor` can be stored in a field.